### ISSUE TITLE
Mini-Epic: Detailed User Search History

### ISSUE THREAD

**Issue by pconrad** [ucsb-cs156-s21/proj-mapache-search#227]
## This story is a "mini-epic" that has been broken into three smaller stories for you. 

This issue should provide some insight on how all three stories work together, as well as show you what the larger goal is, if you're confused about the individual stories. All the work for this issue should be done within the sub-stories though. Don't close this issue unless all of the sub-stories are done.

# User Stories

As a member, each time I do a google search, an entry is made in a log table of google searches performed. This table should probably be a Mongo table (because of the 10K row limit of the free tier of Heroku). The log should include the user id, date time-stamp, and the search query. (#43)

As a member, I can see a list of only my own past search queries in the app. (#41)

As an admin, I can see a list of all past search queries in the app. (#251)

# Acceptance Criteria (across all three stories)

- [ ] There is a new menu item called Search History and that menu item takes us to a Search History Page (this could be a standalone story; to just create a placeholder page)
- [ ] On the Search History Page, there is a sortable table with user id, date time-stamp, and the search query.
- [ ] For Members, the table only includes entries for the current user
- [ ] For Admins, the table shows entries for all users

# Implementation Todos

## Front end:

FIrst front end story:
- [ ] Search History item added to menu bar
- [ ] New page component `SearchHistoryPage` added to `App.js` at an appropriate route
- [ ] New page component `SearchHistoryPage` added to Storybook and has basic test coverage

Second front end story (depends on backend work).
- [ ]  `SearchHistoryPage` retrieves and displays data from backend.

## Back end:

Implementing these backend endpoints with unit tests and exploratory testing through Swagger could be a separate story implemented in parallel, any time before the second front end story.

- [ ] There is a document class that represents one document in the MongoDB Collection for SearchHistory.  Note that this is NOT the same as the existing `Search` entity; that table only stores how many times a particular term was searched, and has a row per search term.  It does NOT have a row per search.  And, that is NOT a capped collection.
- [ ] There is a MongoRepository class for the SearchHistoryRepository.  This should be set up as a capped collection in Mongo.
- [ ] There is a POST api route in a SearchHistoryController to add a row to the SearchHistory MongoDB collection
- [ ] There is a GET api route in a SearchHistoryController that gets all of the data ... all of it for Admins, and just the data for the current user for members.   No results for anyone that isn't a member.
